src/libsyntax/tokenstream.rs

```diff
@@ -820,98 +929,108 @@ impl Index<usize> for TokenStream {
     type Output = TokenTree;
 
     fn index(&self, index: usize) -> &TokenTree {
```

Comment:
I think that any flattening should be obscured from the user, so indexing would have to account for it. I don't mind indexing being slow, if it is documented, etc. Given the syntax etc., it seems unlikely that people would iterate over tokens using indexing.